File: services/common/src/main/java/com/example/common/validator/FieldValidator.java
package com.example.common.validator;

import java.util.function.Predicate;

public class FieldValidator {

    protected static Predicate<String> accountId = s -> new StringValidator(s)
            .notNull()
            .sizeEqualTo(36)
            .isValid();

    protected static Predicate<String> transactionId = s -> new StringValidator(s)
            .notNull()
            .sizeEqualTo(36)
            .isValid();

    protected static Predicate<String> symbol = s -> new StringValidator(s)
            .notNull()
            .notEmpty()
            .sizeGreaterOrEqualTo(1)
            .sizeLessOrEqualTo(6)
            .isValid();

    protected static Predicate<Integer> limit = n -> new IntegerValidator(n)
            .notNull()
            .greaterThanZero()
            .isValid();
}

File: services/common/src/main/kotlin/com/example/common/validator/FieldValidator.kt
package com.example.common.validator


open class FieldValidator {

    companion object {
        var symbol= { s: String ->
            StringValidator(s)
                    .notEmpty()
                    .sizeGreaterOrEqualTo(1)
                    .sizeLessOrEqualTo(6)
                    .isValid()
        }

        var limit = { n: Int ->
            IntegerValidator(n)
                    .greaterThanZero()
                    .isValid()
        }

        var UUID = { s: String ->
            StringValidator(s)
                    .sizeEqualTo(36)
                    .isValid()
        }
    }

}